### PR TITLE
http_request http fix

### LINKED ISSUE #1040: http_request in dev branch is sending corrupted request
**Operating environment/Installation (Hass.io/Docker/pip/etc.):**

ESPHome dev (1.5.0-dev) on 
- Windows 10 (pip install)
- Hassio add-on

**ESP (ESP32/ESP8266, Board/Sonoff):**

ESP8266 nodemcuv2

**Affected component:**

http_request

**Description of problem:**
I tried using the new http_request component via the dev channel. The request is send and "something" is received. However the request seems corrupted, so that it can not be handled. In the esphome log there is only one line related to the component.
Note: I added two additional debugging messages in the http_request component for the HTTP Header and Request parameters and the HTTP Return Code (see the log).
The problem seems not to be plattform dependent. I tried build and deploy via Windows 10 and Hassio add-on (dev).

**Problem-relevant YAML-configuration entries:**
```yaml
http_request:
  useragent: esphome/device
  timeout: 10s

switch:
  - platform: gpio
    pin: 10
    name: "Test http_request"
    on_turn_on:
    - http_request.send: 
        method: GET
        url: http://192.168.179.101:3001/
        headers:
          Content-Type: application/json
        verify_ssl: false
```

**Logs (if applicable):**

ESPHome Log:
```
[07:59:32][I][app:058]: setup() finished successfully!
[07:59:32][I][app:101]: ESPHome version 1.15.0-dev compiled on Feb  8 2020, 07:59:02
...
[09:13:11][D][switch:029]: 'Test http_request' Toggling ON.
[09:13:11][D][switch:045]: 'Test http_request': Sending state ON
[09:13:11][D][http_request:041]: HTTP Header; Name: Content-Type; Value: application/json
[09:13:11][D][http_request:044]: HTTP Request send; Method: GET; Body: ; User Agent: esphome/device
[09:13:11][D][http_request:047]: HTTP Code: 1073653085
[09:13:11][W][http_request:050]: HTTP Request failed; URL: http://192.168.179.101:3001/; Error: connection refused
```

The recieved http request send by esphome:
```
pi@Hassio:~ $ nc -l 3001
▒▒8▒▒▒▒▒R▒▒▒▒▒nc▒▒▒$▒▒1IZ̨̩▒+▒/▒,▒0▒▒▒▒▒#▒'▒$▒(▒  ▒▒
▒▒-▒1▒.▒2▒%▒)▒&▒*▒▒▒▒▒▒▒▒▒▒<=/5▒▒▒
P▒192.168.179.101

                                                                                 PuTTYPuTTYPuTTY
```
I would expect (request send by a browser):
```
pi@Hassio:~ $ nc -l 3001
GET /api/v1/commands/?cmd=toggle HTTP/1.1
Host: 192.168.179.101:3001
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:72.0) Gecko/20100101 Firefox/72.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
Accept-Language: de,en-US;q=0.7,en;q=0.3
Accept-Encoding: gzip, deflate
Connection: keep-alive
Upgrade-Insecure-Requests: 1
Pragma: no-cache
Cache-Control: no-cache
```

**Additional information and things you've tried:**
- Tried differen Content-Types: text/plain as well as application/json
- Tried different Methods (GET, PUT, POST)
- Tried different User Agents

### PR DESCRIPTION
**Pull request by Anonym-tsk**
## Description:
Fixed HTTP requests

**Related issue (if applicable):** fixes https://github.com/esphome/issues/issues/1040

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by Anonym-tsk** [comment]
@OttoWinter

File: esphome/components/http_request/__init__.py [hunk 1]
```diff
@@ -123,6 +122,7 @@ def http_request_action_to_code(config, action_id, template_arg, args):
 
     template_ = yield cg.templatable(config[CONF_URL], args, cg.const_char_ptr)
     cg.add(var.set_url(template_))
+    cg.add(var.set_secure(template_.startswith('https:')))
```
Comment (mnoorenberghe) [inline esphome/components/http_request/__init__.py]: Hello, I thought I would try out your patch as I'm wanting to use `http_request`…

I found that this line doesn't work with lambdas:
> AttributeError: 'LambdaExpression' object has no attribute 'startswith'

**Review by mnoorenberghe — CHANGES_REQUESTED** [review]

File: esphome/components/http_request/__init__.py (hunk 1 above)
Comment (Anonym-tsk) [reply esphome/components/http_request/__init__.py]: Thank you for this. I fixed this case

**Comment by brandond** [comment]
LGTM

**Pull request merged**